### PR TITLE
Add initial GNI provider man page

### PR DESCRIPTION
**Pull request by sungeunchoi**
Still needs work.  Will add link from fi_provider man page later.

@hppritcha @jshimek @jswaro @e-harvey @ztiffany 

Signed-off-by: Sung-Eun Choi sungeunchoi@users.noreply.github.com

### CODE REVIEW

File: man/fi_gni.7.md [hunk 1]
```diff
@@ -0,0 +1,122 @@
...
+The GNI provider run on Cray XC Systems running CLE 5.2 UP02 or
+higher using gcc version 4.9 or higher.
+
+# SUPPORTED FEATURES
+
+The GNI provider supports the following features defined for the
+libfabric API:
+
+*Endpoint types*
+: The provider supports the *FI_EP_MSG* endpoint type.
```
Comment (ztiffany) [inline man/fi_gni.7.md]: We support FI_EP_RDM, not MSG

File: man/fi_gni.7.md [hunk 2]
```diff
@@ -0,0 +1,122 @@
...
+  types: *fi_atomic*, *fi_msg*, *fi_rma*, *fi_tagged*.
+
+*Completion events*
+: The proveders supports *FI_CQ_FORMAT_CONTEXT*, *FI_CQ_FORMAT_MSG*,
+  FI_CQ_FORMAT_DATA* and FI_CQ_FORMAT_TAGGED* with wait objects of type
+  *FI_WAIT_NONE*, *FI_WAIT_FD*, and *FI_WAIT_MUTEX_COND*.
+
+*Modes*
+: The GNI provider requires the following modes: *FI_LOCAL_MR* and
+  *FI_CONTEXT*.
```
Comment (ztiffany) [inline man/fi_gni.7.md]: We don't require FI_CONTEXT any more.

**Comment by ztiffany** [comment]
:+1:

File: man/fi_gni.7.md (hunk 1 above)
Comment (sungeunchoi) [inline man/fi_gni.7.md]: I'm going to blame that type-o on having written that part at 6:30am on the plane.

File: man/fi_gni.7.md (hunk 2 above)
Comment (sungeunchoi) [inline man/fi_gni.7.md]: Got it.

**Pull request merged**